usermods/ULC_Battery_Management/usermod_ulc_batterymanagement.h

```diff
@@ -0,0 +1,279 @@
+#pragma once
+
+#include "wled.h"
+#include "IP5306_I2C.h"
+#include <driver/adc.h>
+
+// the frequency to check battery voltage, 5 seconds
+#ifndef USERMOD_BATTERY_MEASUREMENT_INTERVAL
+  #define USERMOD_BATTERY_MEASUREMENT_INTERVAL 5000
+#endif
+
+#ifndef USERMOD_BATTERY_ADC_PRECISION
+  #define USERMOD_BATTERY_ADC_PRECISION 4095.0f
+#endif
+
+#ifndef USERMOD_BATTERY_ADC_REF
+  #define USERMOD_BATTERY_ADC_REF 3.3f
+#endif
+
+#ifndef USERMOD_BATTERY_ADC_R1
+  #define USERMOD_BATTERY_ADC_R1 8.0f
+#endif
+
+#ifndef USERMOD_BATTERY_ADC_R2
+  #define USERMOD_BATTERY_ADC_R2 2.5f
+#endif
+
+#ifndef USERMOD_BATTERY_MIN_VOLTAGE
+  #define USERMOD_BATTERY_MIN_VOLTAGE 2.6f
+#endif
+
+#ifndef USERMOD_BATTERY_MAX_VOLTAGE
+  #define USERMOD_BATTERY_MAX_VOLTAGE 4.2f
+#endif
+
+#ifndef USERMOD_IP5306_SDA
+  #define USERMOD_IP5306_SDA 32
+#endif
+
+#ifndef USERMOD_IP5306_SCL
```

Comment:
I2C bus is shared among all usermods in 0.14 since #3238.
Please update code to use global I2C pins.